README.md

```diff
@@ -63,6 +63,9 @@ target_link_libraries(minimal_node PRIVATE
   rclcpp::rclcpp
   turtlesim_parameters
 )
+
+install(TARGETS minimal_node turtlesim_parameters
+  EXPORT ${PROJECT_NAME}Targets
```

Comment:
there is a closing parenthesis missing.

Could you explain the difference of 
`EXPORT ${PROJECT_NAME}Targets` here
and 
`EXPORT export_generate_parameter_library_example` from the example/CMakeLists.txt please?

Reply:
Hi,

So the reason why I made these changes is that we tried to package many ros2 packages, and that some failed because of the changes I made in the first place.

The problem with my first implementation was that the generated parameters target was exported automatically by the cmake macro under the name `${PROJECT_NAME}Targets`. Say you had another target, like a library, using the generated parameters one, and exported under a different name than `${PROJECT_NAME}Targets`, cmake would raise an error saying that the generated parameter target was exported multiple times.

So the solution I found was to remove the automatic target export from the macro. So instead of
```cmake
generate_parameters(params_target params_file.yaml)

add_library(my_lib my_lib.cpp)
target_link_libraries(my_lib SHARED params_target)

install(TARGETS my_lib EXPORT WhatheverExportNameYouWant)
ament_export_targets(WhatheverExportNameYouWant HAS_LIBRARY_TARGET)
```

You now need to specify `params_target` in the `TARGETS` list.
This actually make sense : similarly to `add_library`, the `generate_parameters` macro creates a cmake target. It is up to you to export it if you want, and under whatever name you want.

To answer your last question, these are simply names. You can write whatever you want. You could export multiple targets under different names I guess, although in small packages you tend to export all targets under a single name. At least that's what I encountered so far. `${PROJECT_NAME}Targets` will simply expand to `MyProjectTargets` if your cmake project is called `MyProject`.

To conclude, this new implementation requires slightly more work since you need to export the generated parameters target manually if you want to use it outside of your package. But it does not break current implementation in many packages such as `ros2_control`, which is better if you want this path to be released quickly, and backported (no additional work required) to humble.

Do not hesitate if you have more questions or comments.

Reply:
Hi @christophfroehlich 
Is there something else you would like to address regarding this PR ?

Reply:
Sorry, I haven't found time to test the last version. But there is a merge conflict now with the main branch?

Reply:
I rebased the branch on the last version of main. All tests now pass.